### DISCUSSION TITLE
Socket Error - Permission denied on Azure Functions

### DISCUSSION THREAD

**Discussion by Fillo22** [kedacore/keda#1589] [Q&A / Need Help] [answered]
Hi everyone, 
I'm trying to deploy an Azure Function (Event Hub Triggered), but after starting the pod crashes with the following error:
`
System.Net.Sockets.SocketException (13): Permission denied
`

Event Hub is connected via Private Endpoint and all ports are open.
Keda operator is the latest.
Has anyone ever had the same problem?

**Comment by TsuyoshiUshio** [comment, marked as answer]
Hi @Fillo22 
I heard about that. Could you try wget https://<yournamespacename>.servicebus.windows.net/  from your Azure Functions pod? 
This page might be useful. https://docs.microsoft.com/en-us/azure/event-hubs/troubleshooting-guide#troubleshoot-network-related-issues

**Reply by Fillo22** [reply]
Hi @TsuyoshiUshio, finally i found a solution.

Even if an http trigger is not used, the dotnet core runtime tries to expose a reserved port (1 - 1024), however it does not have the permissions to do this, so it fails. To work around the problem I added an environment variable `ENV ASPNETCORE_URLS=http://*:8080` ed I exposed a non-root port.

**Reply by TsuyoshiUshio** [reply]
Great finding. Thank you for sharing.
